internal/locate/region_cache.go

```diff
@@ -1123,9 +1093,18 @@ func (c *RegionCache) findRegionByKey(bo *retry.Backoffer, key []byte, isEndKey
 			c.insertRegionToCache(r, true, true)
 			c.mu.Unlock()
 		}
-	} else if r.checkNeedReloadAndMarkUpdated() {
+	} else if flags := r.resetSyncFlags(needReloadOnAccess | needAsyncReloadReady); flags > 0 {
 		// load region when it be marked as need reload.
-		lr, err := c.loadRegion(bo, key, isEndKey)
+		reloadOnAccess := flags&needReloadOnAccess > 0
+		var (
+			lr  *Region
+			err error
+		)
+		if reloadOnAccess {
+			lr, err = c.loadRegion(bo, key, isEndKey)
+		} else {
+			lr, err = c.loadRegionByID(bo, r.GetID())
+		}
```

Comment:
What's the difference here using by id or by key?

Reply:
Here is the replacement of previous async reload introduced by https://github.com/tikv/client-go/pull/843, which uses `loadRegionByID`. I just try to keep the original implementation but do not very sure about why we use `loadRegionByID` actually. Maybe it's more efficient than `loadRegionByKey`.